src/STW.ProcessingApi.Function/Triggers/ProcessingApiTrigger.cs

```diff
@@ -8,16 +12,27 @@ namespace STW.ProcessingApi.Function.Triggers;
 public class ProcessingApiTrigger
 {
     private readonly ILogger _logger;
+    private readonly IValidator _validator;
 
-    public ProcessingApiTrigger(ILogger<ProcessingApiTrigger> logger)
+    public ProcessingApiTrigger(ILogger<ProcessingApiTrigger> logger, IValidator validator)
     {
         _logger = logger;
+        _validator = validator;
     }
 
     [Function(nameof(ProcessingApiTrigger))]
     public HttpResponseData Run([HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "test")] HttpRequestData request)
     {
         _logger.LogInformation($"{nameof(ProcessingApiTrigger)} function was invoked.");
+        using (var reader = new StreamReader(request.Body, Encoding.UTF8))
+        {
+            var requestBody = reader.ReadToEnd();
```

Comment:
You can change the method to `public async Task<HttpResponseData>` which will then allow you to read the body async which can be simplified to:

```csharp
using var reader = new StreamReader(request.Body);
var requestBody = await reader.ReadToEndAsync();
```

The above away is called a using declaration which was introduced in .NET 8 and is cleaner than the traditional using statement

Reply:
changed as suggested